workflows/models.py

```diff
@@ -663,8 +671,58 @@ def search_input_datasets(args):
         """
         Get the input BAM datasets for this analysis.
         """
-        datasets = tantalus_api.list('sequence_dataset', analysis=args["align_analysis"], dataset_type='BAM')
-        return [dataset['id'] for dataset in datasets]
+
+        filter_lanes = []
+        if args['gsc_lanes'] is not None:
+            filter_lanes.append(args['gsc_lanes'])
+            flowcell_id = args['gsc_lanes']
+        if args['brc_flowcell_ids'] is not None:
+            filter_lanes.append(['{}_{}'.format(args['brc_flowcell_ids'], i+1) for i in range(4)])
+            flowcell_id = args['brc_flowcell_ids']
+
+
+        # Filter for datasets with specific lanes if given
+        if args['gsc_lanes'] is not None or args['brc_flowcell_ids'] is not None:
```

Comment:
this needs to be reworked, querying for flowcell id wont work because its a list of flowcells.  first query for the flowcells based on gsc lanes or brc lanes to build a list of lane pks.  if none of those are given just query for all lanes for the given library.  then query for bam datasets, filtering the ones that dont have the exact subset of lanes